frontend/src/components/Form/Form.js

```diff
@@ -6,73 +6,140 @@ import routePaths from '../../constants/routePaths';
 const Form = (props) => {
   const { formType, handleSubmit, inputsArr } = props;
 
+  const categoriesArr = props.categoriesArr;
+
   const inputsMap = {};
 
-  const oppositeType = formType === 'Register' ? 'Login' : 'Register';
+  const oppositeType =
+    formType === 'Register' ? 'Login' : formType === 'Login' ? 'Register' : '';
+
+  const buttonText =
+    formType === 'New Product' ? 'Create New Product' : formType;
 
   //mapping inputsArr prop to object for use in UseState hook
   inputsArr.forEach((input) => {
-    inputsMap[input] = '';
+    inputsMap[input] = {
+      value: '',
+      isValid: false,
+    };
   });
 
   const [inputs, setInputs] = useState(inputsMap);
 
   //this exists to keep any value change separate from the inputsMap and the state hook
+  //until we are ready to update
   const inputsDestructure = { ...inputs };
 
   const [submitted, setSubmitted] = useState(false);
 
   const onSubmit = (event) => {
     event.preventDefault();
-
     setSubmitted(true);
-
-    for (const prop in inputs) {
-      if (!inputs[prop]) {
+    const inputsResponse = {};
+    for (const key in inputs) {
+      if (!inputs[key].isValid) {
         return;
       }
+      inputsResponse[key] = inputs[key].value;
     }
 
-    handleSubmit(inputs);
+    handleSubmit(inputsResponse);
   };
 
   const onChange = (event) => {
     const { name, value } = event.target;
```

Comment:
you can accomplish this line in the arguments of the method

Reply:
done